tests/main/microk8s-smoke/task.yaml

```diff
@@ -0,0 +1,66 @@
+summary: Smoke test for microk8s
+
+systems:
+  - -amazon-linux-2-* # fails to start service daemon-containerd
+  - -centos-7-*       # doesn't have libseccomp >= 2.4
+  - -centos-8-*       # fails to start service daemon-containerd
+  - -fedora-33-*      # fails to start service daemon-containerd
+  - -fedora-34-*      # fails to start service daemon-containerd
+  - -debian-10-*      # doesn't have libseccomp >= 2.4
+  - -ubuntu-14.04-*   # doesn't have libseccomp >= 2.4
+
+environment:
+    CHANNEL/edge: latest/edge/strict
+
+prepare: |
+    # ensure curl is available (needed for Ubuntu Core)
+    if ! command -v curl; then
+        snap install --devmode --edge test-snapd-curl
+        snap alias test-snapd-curl.curl curl
+    fi
+
+    # The default timeout for the configure hook is 5min - however in the
+    # testsuite this is lowered to 30 seconds. We need to undo this for the
+    # microk8s spread test because it really take a bit to get configured.
+    cp /etc/systemd/system/snapd.service.d/local.conf /etc/systemd/system/snapd.service.d/local.conf.bak
+    sed 's/SNAPD_CONFIGURE_HOOK_TIMEOUT=.*s/SNAPD_CONFIGURE_HOOK_TIMEOUT=180s/g' -i /etc/systemd/system/snapd.service.d/local.conf
+    systemctl daemon-reload
+    systemctl restart snapd.socket
+
+restore: |
+    mv /etc/systemd/system/snapd.service.d/local.conf.bak /etc/systemd/system/snapd.service.d/local.conf
+    systemctl daemon-reload
+    systemctl restart snapd.socket
+
+    snap remove --purge test-snapd-curl
+
+    snap remove --purge microk8s
```

Comment:
nitpick all of these things could be done with individual `tests.cleanup defer ...` statements at corresponding locations and avoid the issue like where if we fail to install test-snapd-curl from the store (i.e. because we hit 408 from the particular spread VM IP address, then we won't have created the .bak file and that will fail and the spread output will become needlessly verbose including "error restoring ..." in addition to the "error preparing ..."